### PR TITLE
[linux]:Implement soft AP APIs for ConnectivityMgr on Linux platform

### LINKED ISSUE #3860: Implement SoftAP APIs within ConnectivityManager on Linux reference platform
#### Problem
Currently, we only support STA APIs within ConnectivityManager on Linux reference platform, network provisioning over IP need put the device under AP mode. 

 #### Proposed Solution
Implement SoftAP APIs within ConnectivityManager on Linux reference platform

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
Currently, we only support STA APIs within ConnectivityManager on Linux reference platform, network provisioning over IP need put the device under AP mode.

 #### Summary of Changes
Implement soft AP APIs for ConnectivityMgr on Linux platform

 Fixes #3860

### CODE REVIEW

**Review by yunhanw-google — APPROVED** [review]

File: src/platform/Linux/ConnectivityManagerImpl.cpp [hunk 1]
```diff
@@ -46,6 +46,179 @@ using namespace ::chip::DeviceLayer::Internal;
 namespace {
 const char kWpaSupplicantServiceName[] = "fi.w1.wpa_supplicant1";
 const char kWpaSupplicantObjectPath[]  = "/fi/w1/wpa_supplicant1";
+
+constexpr uint16_t kWiFi_BAND_2_4_GHZ = 2400;
+constexpr uint16_t kWiFi_BAND_5_0_GHZ = 5000;
+
+static uint16_t Map2400MHz(const uint8_t inChannel)
```
Comment (andy31415) [inline src/platform/Linux/ConnectivityManagerImpl.cpp:53]: static not required: this is already in an anonymous namespace. Here and below.

File: src/platform/Linux/ConnectivityManagerImpl.h [hunk 2]
```diff
@@ -126,6 +126,17 @@ class ConnectivityManagerImpl final : public ConnectivityManager,
     bool _IsWiFiStationProvisioned();
     void _ClearWiFiStationProvision();
     bool _CanStartWiFiScan();
+
+    WiFiAPMode _GetWiFiAPMode(void);
```
Comment (andy31415) [inline src/platform/Linux/ConnectivityManagerImpl.h:130]: please remove the void in `(void)` - we had a clang tidy pass that already removed it everywhere else.

**Review by andy31415 — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

File: src/platform/Linux/ConnectivityManagerImpl.h (hunk 2 above)
Comment (yufengwangca) [reply src/platform/Linux/ConnectivityManagerImpl.h]: Acked

File: src/platform/Linux/ConnectivityManagerImpl.cpp (hunk 1 above)
Comment (yufengwangca) [reply src/platform/Linux/ConnectivityManagerImpl.cpp]: Good catch! acked.

**Review by jelderton — APPROVED** [review]

**Pull request merged**